packages/core/lib/combo/logic/world-transform.ts

```diff
@@ -1035,6 +1035,22 @@ export class LogicPassWorldTransform {
       this.removeLocations(['OOT Hatch Chicken', 'OOT Hatch Pocket Cucco']);
     }
 
+    /* Handle Diving Game Rupees */
+    if (!settings.divingGameRupeeShuffle) {
```

Comment:
This should go before `makePools()` is called, that way you don't have to remove items, only locations (see how grass & pots handle that)

Reply:
Done :)